### ISSUE TITLE
Warnings with gcc-8.1.0

### ISSUE THREAD

**Issue by adydychk** [Intel-Media-SDK/MediaSDK#359]
Build environment:
Ubuntu Server 18.04
Compiler: gcc 8.1.0 (installed from repositories)

Build: without X11, gtests, OpenCL

Used cmake options:
```
cmake --no-warn-unused-cli -Wno-dev -G "Unix Makefiles" -DCMAKE_C_FLAGS_RELEASE="-O2 -Wformat -Wformat-security -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong" -DCMAKE_CXX_FLAGS_RELEASE="-O2 -Wformat -Wformat-security -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong" ..cmake --no-warn-unused-cli -Wno-dev -G "Unix Makefiles" -DCMAKE_C_FLAGS_RELEASE="-O2 -Wformat -Wformat-security -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong" -DCMAKE_CXX_FLAGS_RELEASE="-O2 -Wformat -Wformat-security -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong" ..
```
Output of cmake:
```
Not searching for unused variables given on the command line.
-- The C compiler identification is GNU 8.1.0
-- The CXX compiler identification is GNU 8.1.0
-- Check for working C compiler: /usr/bin/gcc-8
-- Check for working C compiler: /usr/bin/gcc-8 -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/g++-8
-- Check for working CXX compiler: /usr/bin/g++-8 -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- No build type selected, default to release
-- Intel OpenCL SDK was not found (optional). The following will not be built: rotate_opencl plugin.
-- CMAKE_INSTALL_PREFIX=/opt/intel/mediasdk
-- Intel(R) Media SDK was found here /localdisk/tmp/MediaSDK-gcc8
-- Enabling API 1.27 feature set with flags
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1")
-- Checking for module 'libva>=0.33'
--   Found libva, version 1.1.0
-- Checking for module 'libdrm'
--   Found libdrm, version 2.4.91
-- Checking for module 'libva-drm>=0.33'
--   Found libva-drm, version 1.1.0
-- Checking for module 'x11'
--   No package 'x11' found
-- Checking for module 'libva-x11>=0.33'
--   No package 'libva-x11' found
-- x11 modules not found (optional). Samples will be built without x11 support
-- Google tests libraries and headers were not found! Build GTest and install to /usr/local.
-- Global Configuration of Targets
-- Target Architecture to compile: sse4.2
-- CMAKE_INSTALL_PREFIX=/opt/intel/mediasdk
-- MFX_PLUGINS_DIR=/opt/intel/mediasdk/plugins
-- MFX_MODULES_DIR=/opt/intel/mediasdk/lib
-- Checking for module 'libdrm'
--   Found libdrm, version 2.4.91
-- Checking for module 'pciaccess'
--   Found pciaccess, version 0.14
-- Google tests was not found (optional), the following will not be built: test_monitor.
-- Intel OpenCL SDK was not found (optional). The following will not be built: rotate_opencl plugin.

********************** Media SDK Configuration Summary **********************
MediaSDK:
  API (request to use this version)       :
  API_VERSION                             : 1.27
  API_FLAGS                               :
Cmake:
  CMAKE_VERSION                           : 3.10.2
  CMAKE_GENERATOR                         : Unix Makefiles
  CMAKE_BUILD_TOOL                        : /usr/bin/make
Target:
  CMAKE_SYSTEM_NAME                       : Linux
  CMAKE_SYSTEM_VERSION                    : 4.15.0-23-generic
  CMAKE_SYSTEM_PROCESSOR                  : x86_64
General:
  CMAKE_BUILD_TYPE                        : release
  CMAKE_TOOLCHAIN_FILE                    :
  CMAKE_C_COMPILER                        : /usr/bin/gcc-8
  CMAKE_CXX_COMPILER                      : /usr/bin/g++-8
  Building architecture                   : 64-bit
Flags:
  CMAKE_C_FLAGS                           : -msse4.2 -m64 -g  -pipe -fPIC -Wall -Wformat -Wformat-security -Wno-deprecated-declarations -Wno-unknown-pragmas -Wno-unused
  CMAKE_CXX_FLAGS                         : -msse4.2 -std=c++11 -m64 -g  -pipe -fPIC -Wall -Wformat -Wformat-security -Wnon-virtual-dtor -Wno-deprecated-declarations -Wno-unknown-pragmas -Wno-unused
Release flags:
  CMAKE_C_FLAGS_RELEASE                   : -O2 -Wformat -Wformat-security -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong
  CMAKE_CXX_FLAGS_RELEASE                 : -O2 -Wformat -Wformat-security -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong
Debug flags:
  CMAKE_C_FLAGS_DEBUG                     : -g -D_DEBUG
  CMAKE_CXX_FLAGS_DEBUG                   : -g -D_DEBUG
Install:
  CMAKE_INSTALL_RREFIX                    :
*****************************************************************************
-- Configuring done
-- Generating done
-- Build files have been written to: /localdisk/tmp/MediaSDK-gcc8/build
```

Log of building (with warnings):
[Ubuntu1804log.txt](https://github.com/Intel-Media-SDK/MediaSDK/files/2146236/Ubuntu1804log.txt)

**Comment by dvrogozh** [comment]
@chivakker: this most probably will strike you building ClearLinux. Please, be aware. We are working on resolution.

**Comment by dvrogozh** [comment]
@mgonchar : could you, please, look into and fix gcc-8.1 warnings in FEI part?

**Comment by mgonchar** [comment]
@dvrogozh looked through the log. Looks like gcc-8.1 doesn't like our common practice of memsetting POD structures right after creation. Will add default initializers for them

**Comment by dvrogozh** [comment]
@mgonchar : yep, it does not. Thank you. FYI: CI for gcc-8.1 will be ready and enabled (if all is ok) today/tomorrow (without -Werror). As a next step we need 0 warnings level on gcc-8.1 to use -Werror. Hence, we need to fix this bug...

**Comment by dvrogozh** [comment]
PR #405 should fix most of the warnings except these 2 in sample_hevc_fei_abr:
```
/localdisk/tmp-drogozh/MediaSDK/samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp: In member function ‘BS_HEVC2::TU* BS_HEVC2::SDParser::parseTT(BS_HEVC2::CU&, BS_HEVC2::TU&, BS_HEVC2::TU&, Bs16u)’:
/localdisk/tmp-drogozh/MediaSDK/samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:1190:34: error: ‘void* memset(void*, int, size_t)’ clearing an object of non-trivial type ‘struct BS_HEVC2::TU’; use assignment or value-initialization instead [-Werror=class-memaccess]
         memset(&tu, 0, sizeof(tu));
                                  ^
In file included from /localdisk/tmp-drogozh/MediaSDK/samples/sample_hevc_fei_abr/include/dso/hevc2_parser.h:26,
                 from /localdisk/tmp-drogozh/MediaSDK/samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:21:
/localdisk/tmp-drogozh/MediaSDK/samples/sample_hevc_fei_abr/include/dso/hevc2_struct.h:800:8: note: ‘struct BS_HEVC2::TU’ declared here
 struct TU
        ^~
In file included from /localdisk/tmp-drogozh/MediaSDK/samples/sample_hevc_fei_abr/include/dso/hevc2_parser.h:24,
                 from /localdisk/tmp-drogozh/MediaSDK/samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:21:
/localdisk/tmp-drogozh/MediaSDK/samples/sample_hevc_fei_abr/include/dso/bs_mem+.h: In instantiation of ‘BS_MEM::MemObj<T>::MemObj(unsigned int, bool) [with T = BS_HEVC2::TU]’:
/localdisk/tmp-drogozh/MediaSDK/samples/sample_hevc_fei_abr/include/dso/bs_mem+.h:158:25:   required from ‘T* BS_MEM::Allocator::alloc(void*, unsigned int) [with T = BS_HEVC2::TU]’
/localdisk/tmp-drogozh/MediaSDK/samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:228:61:   required from here
/localdisk/tmp-drogozh/MediaSDK/samples/sample_hevc_fei_abr/include/dso/bs_mem+.h:56:19: error: ‘void* memset(void*, int, size_t)’ clearing an object of non-trivial type ‘struct BS_HEVC2::TU’; use assignment or value-initialization instead [-Werror=class-memaccess]
             memset(m_obj, 0, sizeof(T) * count);
             ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /localdisk/tmp-drogozh/MediaSDK/samples/sample_hevc_fei_abr/include/dso/hevc2_parser.h:26,
                 from /localdisk/tmp-drogozh/MediaSDK/samples/sample_hevc_fei_abr/src/dso/hevc2_ssdata.cpp:21:
/localdisk/tmp-drogozh/MediaSDK/samples/sample_hevc_fei_abr/include/dso/hevc2_struct.h:800:8: note: ‘struct BS_HEVC2::TU’ declared here
 struct TU
        ^~
```
@dmitryermilov, @mgonchar : I need your help to fix these!

**Comment by dvrogozh** [comment]
Ok, https://github.com/Intel-Media-SDK/MediaSDK/pull/408 was just merged and I think this is the last piece of gcc-8.1 warnings puzzle. @adydychk : please, try to enable -Werror for gcc-8.1 build ASAP.

**Comment by dvrogozh** [comment]
This is fixed, CI switched to have -Werror with gcc-8.1 compiler.
